src/ngx_http_lua_headers_in.c

```diff
@@ -698,6 +693,11 @@ ngx_http_lua_set_input_header(ngx_http_request_t *r, ngx_str_t key,
     }
 #endif
 
+    if (r->headers_out.status == 400 || r->headers_in.headers.last == NULL) {
```

Comment:
Is there a reason this test is `||` instead of `&&` . 

Earlier the test was checking for `r->headers_in.headers.last == NULL` inferring that this meant a 400 bad request (at least from the comment). 

Now we are inferring the same thing, but if *either* `r->headers_out.status == 400` *OR* `r->headers_in.headers.last == NULL`. 

We have expanded the scope of the cases we would earlier return `NGX_OK` (assuming there are cases where `out_status == 400 ` and `headers.last != NULL`

LGTM, other that the question above.